### PR TITLE
add `kubernixos build` and move kubeval to build phase

### PR DESCRIPTION
**Pull request by johanot**
- refactor: module changes + drop vendored kubeval-schema
- added: kubernixos build subcommand

### CODE REVIEW

**Review by srhb — DISMISSED** [review]

**Review by srhb — COMMENTED** [review]
~What happens here if a json string includes a single quote?~ didn't get this attached to code somehow. Fun

File: lib/kubernixos.nix [hunk 1]
```diff
@@ -19,26 +20,30 @@ let
...
+          kind = "List";
+          items = lib.attrValues items;
       };
-
   };
+
+  build =
+      pkgs.runCommand "kubernixos-${kubernixos}" { nativeBuiltInputs = [pkgs.kubeval cfg.schemas]; } ''
+          mkdir -p $out
+          ${lib.concatStringsSep "\n" (lib.mapAttrsToList (n: i: "echo '${builtins.toJSON i}' >$out/${n}.json") items)}
```
Comment (srhb) [inline lib/kubernixos.nix]: What happens here if a json string contains a single quote? Accept build failure and move on?

Comment (srhb) [reply lib/kubernixos.nix]: Alternatively one could build `writeText` each json snippet and cat them together, or create both the individuals and the sum in Nix before writing it out. No reason to go through bash strings.

Comment (johanot) [reply lib/kubernixos.nix]: Have a look at: a245a1d4db17ca810cbc4c595eb5f8c2d26b04df

Comment (srhb) [reply lib/kubernixos.nix]: :ok_hand:

**Review by srhb — APPROVED** [review]

**Pull request merged**